### PR TITLE
feat(models): replace boilerplate by lombok

### PR DESCRIPTION
**Pull request by michel-barret**
## Description

Replace generated code for getter, setter, toString, no args constructor, all fields constructor, equals & hashCode by lombok annotations.

It's more declarative and more maintainable

### CODE REVIEW

File: gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/UserMongo.java [hunk 1]
```diff
@@ -26,6 +29,10 @@
  * @author David BRASSELY (david.brassely at graviteesource.com)
  * @author GraviteeSource Team
  */
+@Setter
+@Getter
+@ToString
```
Comment (JedrzejJanasiak) [inline gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/UserMongo.java:34]: For ApiKeyMongo.java you included only a few fields to match the current toString() method. Here you went with the default ToString annotation. I'm ok with that. I'm just wondering if that was on purpose or you just missed that :)

File: gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/ApiKeyMongo.java [hunk 2]
```diff
@@ -31,6 +33,8 @@
  * @author David BRASSELY (david.brassely at graviteesource.com)
  * @author GraviteeSource Team
  */
+@ToString(of = { "key", "expireAt", "revoked", "revokedAt" })
```
Comment (JedrzejJanasiak) [inline gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/ApiKeyMongo.java]: From what I see in the Lombok docs there is a new way of defining included fields. The one you use is old-fashioned and may be deprecated in time. How about switching to the new one? Wdyt? 

https://projectlombok.org/features/ToString  
image

File: gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/ApiKeyMongo.java [hunk 3]
```diff
@@ -31,6 +33,8 @@
...
+@EqualsAndHashCode(of = { "id" })
```
Comment (JedrzejJanasiak) [inline gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/ApiKeyMongo.java]: The same with the @EqulasAndHashCode actually: 

https://projectlombok.org/features/EqualsAndHashCode
image

**Review by JedrzejJanasiak — CHANGES_REQUESTED** [review]

File: gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/UserMongo.java (hunk 1 above)
Comment (michel-barret) [reply gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/UserMongo.java:34]: When I see something like API KEY I imagine that we don't want log a critical data, but your are right, the critical data is already in toString() so I remove the selecting of fields.

File: gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/ApiKeyMongo.java (hunk 2 above)
Comment (michel-barret) [reply gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/ApiKeyMongo.java]: I remove the selection of field https://github.com/gravitee-io/gravitee-api-management/pull/8697#discussion_r1723216629

File: gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/ApiKeyMongo.java (hunk 3 above)
Comment (michel-barret) [reply gravitee-apim-repository/gravitee-apim-repository-mongodb/src/main/java/io/gravitee/repository/mongodb/management/internal/model/ApiKeyMongo.java]: Oh I don't know it, it's fixed

**Review by JedrzejJanasiak — APPROVED** [review]

**Pull request merged**